src/coreclr/jit/forwardsub.cpp

```diff
@@ -636,7 +636,7 @@ bool Compiler::fgForwardSubStatement(Statement* stmt)
     // If fwdSubNode is an address-exposed local, forwarding it may lose optimizations.
     // (maybe similar for dner?)
     //
-    if (fwdSubNode->IsLocal())
+    if (fwdSubNode->OperIs(GT_LCL_VAR))
```

Comment:
Is the check for `GT_LCL_VAR` still necessary?

Reply:
The comment above states:
```
If fwdSubNode is an address-exposed local, forwarding it may lose optimizations.
```
Forwarding address-exposed things "poisons" the whole tree with `GTF_GLOB_REF`, and that can have negative downstream consequences.

For local fields we're apparently better off doing it anyway, so I just trust @AndyAyersMS's initial analysis here, that for `LCL_VAR` it doesn't.

Reply:
Ah ok, I had not noticed that this was not for correctness.